frontend/catalyst/jax_primitives.py

```diff
@@ -1339,29 +1346,31 @@ def _qinst_lowering(
         assert ir.OpaqueType(qubit.type).dialect_namespace == "quantum"
         assert ir.OpaqueType(qubit.type).data == "bit"
 
-    float_params = []
-    for p in params:
+    float_dyn_params = []
+    for p in dyn_params:
         p = safe_cast_to_f64(p, op)
         p = extract_scalar(p, op)
 
         assert ir.F64Type.isinstance(
             p.type
         ), "Only scalar double parameters are allowed for quantum gates!"
 
-        float_params.append(p)
+        float_dyn_params.append(p)
 
     ctrl_values_i1 = []
     for v in ctrl_values:
         p = TensorExtractOp(ir.IntegerType.get_signless(1), v, []).result
         ctrl_values_i1.append(p)
 
+    params_attr = ir.ArrayAttr.get([ir.FloatAttr.get_f64(val) for val in static_params])
+
     name_attr = ir.StringAttr.get(op)
     name_str = str(name_attr)
     name_str = name_str.replace('"', "")
 
     if name_str == "MultiRZ":
-        assert len(float_params) == 1, "MultiRZ takes one float parameter"
-        float_param = float_params[0]
+        assert len(float_dyn_params) <= 1, "MultiRZ takes at most one dynamic float parameter"
```

Comment:
Should we check that, if the angle is not dynamic, then the static path must contain the param? i.e. can't both be empty

Reply:
And also can't both be non-empty, i.e. these two must coexist in "XOR" fashion

Reply:
I think the mutual exclusiveness of static and dynamic params is ensured when binding, but I will add an assertion just in case

Reply:
Some update on this. I think it is actually not a true xor relationship. Static parameters should not be present when having dyn params but we can have no static params and no dyn params as well, e.g. in Hadamard

Reply:
@mehrdad2m correct!